### PR TITLE
Feature/create income type

### LINKED ISSUE #21: Create Model for IncomeType
An IncomeType is a channel for a user to get money some budgets are expected to make incomes this will be advantageous then however when a budget isn't expected to make an income this will be used to 'load' the accounts.
They have these fields 
- name: name of this IncomeType 
- money_log_id: id of the MoneyLog this income type belongs to
- balance: balance loaded into this income type

A model will need to 

- [ ] Create an IncomeType when name and MoneyLog are provided 
- [ ] Update an IncomeType's Balance 
- [ ] Update an IncomeType's name

### PR DESCRIPTION
**Pull request by WanjikuMac**
Fixes #21 

## Proposed Changes
Create the Income type model
  -
  -
  -

### CODE REVIEW

File: test/lib/spender/using/income_type.exs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  @valid_income_type %{name: "Business", balance: "1500.50"}
+  @invalid_income_type %{}
+
+  describe "Income_type changeset" do
+    test "Is valid when provided the right attributes" do
+      changeset = IncomeType.changeset(%IncomeType{}, @valid_income_type)
+      assert changeset.valid?
+    end
+
+    test "Is invalis when provided with the wrong attributes" do
```
Comment (zacck-zz) [inline test/lib/spender/using/income_type.exs]: This test has a spelling error in its name

File: lib/spender/using/income_type.ex [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    and proceeds to match the parameters in the map to the schema above
+  """
+  @spec changeset(IncomeType.t(), map()) :: Ecto.Changeset.t()
+  def changeset(incometype, attrs \\ %{}) do
+    incometype
+    |> cast(attrs,[:name,:balance])
+    |> validate_required([:name,:balance])
+  end
+
+  @spec create_changeset(Budget.t(), map()) :: Ecto.Changeset.t()
```
Comment (zacck-zz) [inline lib/spender/using/income_type.ex]: This function needs a documentation attribute

File: priv/repo/migrations/20180710193411_create_income_type.exs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+defmodule Spender.Repo.Migrations.CreateIncomeType do
+  use Ecto.Migration
+  @moduledoc"""
+  This module hold the database schema for income type table
+"""
```
Comment (zacck-zz) [inline priv/repo/migrations/20180710193411_create_income_type.exs]: please run mix format for this file

**Review by zacck-zz — CHANGES_REQUESTED** [review]
@WanjikuMac  good work on this! One thing this is an issue, shouldn't the IncomeType be in the `Planning` context since people will be recording incomes as part of the planning process, what do you think.?

Also please consider running `mix format` for the files you change, will automate this soon though

File: test/lib/spender/using/income_type.exs (hunk 1 above)
Comment (WanjikuMac) [reply test/lib/spender/using/income_type.exs]: Noted.

**Review by zacck-zz — APPROVED** [review]

**Comment by zacck-zz** [comment]
@WanjikuMac  thank you so much for your work

**Pull request merged**